src/wasm-as.cpp

```diff
@@ -35,6 +36,11 @@ int main(int argc, const char *argv[]) {
                 o->extra["output"] = argument;
                 Colors::disable();
               })
+      .add("--metrics", "", "prints binary metrics",
```

Comment:
The metrics code needs to be interleaved with the serialization code. I don't see how this could be moved to a pass.